### ISSUE TITLE
`deploy.sh` script fails with an invalidly formatted Docker image tag on forked repositories without git tags

### ISSUE THREAD

**Issue by tutebatti** [UniStuttgart-VISUS/damast#196]
Running the script `deploy.sh` with the following command:

`./deploy.sh -p -r -l ubuntu -H 192.168.178.52 -u 998 -g 998 -a -n`

I get the following error:

`invalid argument "damast:-production" for "-t, --tag" flag: invalid reference format`

Is there maybe a typo in the script, i.e. a missing space after the colon after `damast?

**Comment by mfranke93** [comment]
- Did you run this with the unmodified script from the repository? If not, what does `git diff deploy.sh` output?
- What does `git status` output?
- What does `git show HEAD` output?
- What does `git describe --tags` output?
- What is the output of `bash -x deploy.sh -p -r -l ubuntu -H 192.168.178.52 -u 998 -g 998 -a -n`?

**Comment by tutebatti** [comment]
> * Did you run this with the unmodified script from the repository? If not, what does `git diff deploy.sh` output?

I ran it both with a modified version and the regular one. These are the differences of the modified version (`deploy.sh.backup`):

```bash
$ diff deploy.sh deploy.sh.backup 
80c80
< basedir="/www-testing"
---
> basedir="/www"
83c83
< service="damast-testing"
---
> service="damast"
92c92
< www_user_id=997
---
> www_user_id=1001
```

> * What does `git status` output?

```bash
$ git status
On branch feature/better-logos
Your branch is ahead of 'origin/main' by 1 commit.
  (use "git push" to publish your local commits)

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        deploy.sh.backup
        deploy.sh.bak

nothing added to commit but untracked files present (use "git add" to track)
```

> * What does `git show HEAD` output?

```bash
$ git show HEAD
commit 15c0ae2d94bd23c7f37c2f57bda87d9fc4a08e7f (HEAD -> feature/better-logos, origin/feature/better-logos)
Author: tutebatti <florian-jaeckel@gmx.de>
Date:   Wed May 17 09:13:44 2023 +0200

    Use stable wikimedia URLs for logos on homepage

diff --git a/damast/root/templates/root/index.html b/damast/root/templates/root/index.html
index 69525d1..b02f119 100644
--- a/damast/root/templates/root/index.html
+++ b/damast/root/templates/root/index.html
@@ -91,16 +91,16 @@
         <img src="https://upload.wikimedia.org/wikipedia/commons/c/ce/Huberlin-logo.svg" alt="Logo Humboldt University Berlin">
       </a>
       <a class="logo" href="https://www.vis.uni-stuttgart.de/projekte/dhimmis-and-muslims/">
-        <img src="https://www.vis.uni-stuttgart.de/system/modules/de.stuttgart.uni.v3.basics/resources/img/svg/logo-de.svg" alt="Logo University of Stuttgart">
+        <img src="https://commons.wikimedia.org/wiki/File:Uni_stuttgart_logo.svg" alt="Logo University of Stuttgart">
       </a>
       <a class="logo logo--md" href="https://www.geschichte.uni-frankfurt.de/87681539/dhimmis_and_muslims">
         <img src="https://upload.wikimedia.org/wikipedia/de/f/f0/Goethe-Logo.svg" alt="Logo Goethe University">
       </a>
       <a class="logo" href="https://portal.volkswagenstiftung.de/search/projectDetails.do?ref=93252">
-        <img class="white" src="https://www.volkswagenstiftung.de/profiles/vwst/theme/volkswagen/images/logo.svg" alt="Logo VolkswagenStiftung">
+        <img class="white" src="https://upload.wikimedia.org/wikipedia/commons/2/21/Logo_Volkswagenstiftung.svg" alt="Logo VolkswagenStiftung">
       </a>
       <a class="logo" href="https://www.dfg.de/">
-        <img class="white" src="https://www.geschichte.uni-frankfurt.de/100952108/dfg_logo_schriftzug_blau_foerderung_en.jpg" alt="LogoDFG">
+        <img class="white" src="https://upload.wikimedia.org/wikipedia/commons/8/86/DFG-logo-blau.svg" alt="LogoDFG">
       </a>
     </div>
```

> * What does `git describe --tags` output?

```bash
$ git describe --tags
fatal: No names found, cannot describe anything.
```

Is that the problem?

**Comment by mfranke93** [comment]
> I ran it both with a modified version and the regular one. These are the differences of the modified version (deploy.sh.backup):

Those changes are irrelevant if you use the `-p` and `-u 998` flags anyways. (Why is it 997 in the script? And why is the group ID not overridden?) But that should not make any difference for this issue.
As a side note, if you want a more "git-like" diff, you can use `git diff --no-index <file A> <file B> :smiley: 

 > Is that the problem?

Yes. You are on a fork right now, right? I just forked as well to test it out. Apparently forking a repo will not copy the upstream tags (which seems very weird to me...).

I will add some documentation to the README regarding forks, and also put a safeguard in the deploy script. For now, what you need to do in your repository is:
``` bash
$ git remote add upstream git@github.com:UniStuttgart-VISUS/damast.git  # or https:// URL, whatever you used before
$ git fetch upstream --tags
```
Then, `git describe --tags *should* output something along the lines of `v1.1.8-1-g1234567`. Please let me know if that fixes it.

**Comment by mfranke93** [comment]
> Then, `git describe --tags` should output something along the lines of `v1.1.8-1-g1234567`. Please let me know if that fixes it.

I got exactly the same issue on my fresh checkout from my fork, so I went ahead and merged it (#198).

**Comment by tutebatti** [comment]
Thanks! The steps worked as described.
